### PR TITLE
[admin] Introduce change own password function

### PR DESCRIPTION
**Pull request by kkana6624**
# Changes

* Added change own password function
* Fixed change password function for super admin
* Do mix format

# Review Perspective

* Whether the content of the implementation is appropriate
* Whether UI behaviour is appropriate
* Make sure unit tests are properly implemented
* Please check if the modified password change function for super admin is appropriate

# Questions

* I have given the password input field a `required` attribute, but should I implement a separate implementation to display `cant be blank`?

### CODE REVIEW

**Comment by kkana6624** [comment]
@kuroda 
Sorry for the delay in response.
Please review.

**Comment by kkana6624** [comment]
ボタンの挙動

**Comment by kkana6624** [comment]
キャンセル

**Comment by kkana6624** [comment]
パスワード変更操作

**Comment by kkana6624** [comment]
@kuroda 
I apologize for the delayed response. I have corrected the points you mentioned in the previous review.

There is one part where I am unsure about the implementation method, and it is behaving strangely. Regarding the toggle of the user menu display, I was able to implement the action to hide the menu when clicking outside the menu using phx-click-away. 

However, when clicking the username button again, the menu toggle process is executed twice. 

If you have a good implementation method, please let me know.

File: lib/live_web_server_web/components/layouts/app.html.heex [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <header class="bg-gray-800 text-white p-4">
   <%= if @current_administrator do %>
     <nav phx-hook="ShowModalSignOut" id="live-web-server-header" class="flex ml-4">
-      <div class="flex flex-wrap content-center">
+      <div class="flex flex-wrap content-center" phx-click-away="hide_menu">
```
Comment (kuroda) [inline lib/live_web_server_web/components/layouts/app.html.heex]: Remove `phx-click-away="hide_menu"`

File: lib/live_web_server_web/components/layouts/app.html.heex [hunk 2]
```diff
@@ -20,12 +20,32 @@
         </div>
       </div>
 
-      <div class="flex flex-grow justify-end">
-        <button id="open-sign-out-modal" class="btn btn-neutral">
-          <%= "Sign out" %>
+      <div class="flex flex-grow justify-end" phx-click-away="hide_menu">
```
Comment (kuroda) [inline lib/live_web_server_web/components/layouts/app.html.heex]: Remove `phx-click-away="hide_menu"`.

File: lib/live_web_server_web/components/layouts/app.html.heex [hunk 3]
```diff
@@ -20,12 +20,32 @@
...
+        <button class="btn btn-neutral" phx-click="toggle_menu">
```
Comment (kuroda) [inline lib/live_web_server_web/components/layouts/app.html.heex]: Add `phx-click-away="hide_menu"` here.

**Review by kuroda — CHANGES_REQUESTED** [review]
I add some comments as an answer to your question.

**Comment by kkana6624** [comment]
@kuroda 
Thank you for your review comments. 
After making the corrections, it now works as expected.

**Comment by kuroda** [comment]
@kkana6624 

Once inside the container, run `mix format`. From now on, run it every time before you commit.

**Comment by kkana6624** [comment]
@kuroda 
I ran mix format.
For reference, could you tell me if there is a difference in meaning between
 running `mix format` on the **WSL console** and running `mix format` **inside a container?**

**Comment by kuroda** [comment]
@kkana6624 

> For reference, could you tell me if there is a difference in meaning between
> running mix format on the WSL console and running mix format inside a container?

Usually, there may be no difference. I have considered the possibility of different Elixir versions inside and outside the container.

File: assets/js/app.js [hunk 4]
```diff
@@ -18,8 +18,8 @@
 // Include phoenix_html to handle method=PUT/DELETE in forms and buttons.
 import "phoenix_html"
 // Establish Phoenix Socket and LiveView configuration.
-import {Socket} from "phoenix"
-import {LiveSocket} from "phoenix_live_view"
+import { Socket } from "phoenix"
```
Comment (kuroda) [inline assets/js/app.js]: Please leave this as it is. `{Socket}`

Are these modifications due to some tool?

File: assets/js/app.js [hunk 5]
```diff
@@ -35,14 +35,14 @@ Hooks.ShowModalSignOut = {
     const openButton = document.getElementById("open-sign-out-modal")
     const closeButton = document.getElementById("close-sign-out-modal")
 
-    openButton.addEventListener("click", e => {
+    openButton.addEventListener("click", (e) => {
```
Comment (kuroda) [inline assets/js/app.js]: Do not add parentheses.

File: assets/js/app.js [hunk 6]
```diff
@@ -35,14 +35,14 @@ Hooks.ShowModalSignOut = {
...
       e.stopPropagation()
-      document.getElementById("sign-out-dialog").showModal();
+      document.getElementById("sign-out-dialog").showModal()
     })
 
-    closeButton.addEventListener("click", e => {
+    closeButton.addEventListener("click", (e) => {
```
Comment (kuroda) [inline assets/js/app.js]: Do not add parentheses.

File: assets/js/app.js [hunk 7]
```diff
@@ -68,32 +68,34 @@ Hooks.ShowPassword = {
         changePasswordInput.setAttribute("type", type)
       }
     })
-  }
+  },
```
Comment (kuroda) [inline assets/js/app.js]: Do not add a comma here.

File: assets/js/app.js [hunk 8]
```diff
@@ -68,32 +68,34 @@ Hooks.ShowPassword = {
...
 }
 
-let csrfToken = document.querySelector("meta[name='csrf-token']").getAttribute("content")
+let csrfToken = document
+  .querySelector("meta[name='csrf-token']")
+  .getAttribute("content");
```
Comment (kuroda) [inline assets/js/app.js]: Please leave this as it is.

File: assets/js/app.js [hunk 9]
```diff
@@ -68,32 +68,34 @@ Hooks.ShowPassword = {
...
 let liveSocket = new LiveSocket("/live", Socket, {
   hooks: Hooks,
   longPollFallbackMs: 2500,
-  params: {_csrf_token: csrfToken}
+  params: { _csrf_token: csrfToken },
```
Comment (kuroda) [inline assets/js/app.js]: Please leave this as it is.

File: assets/js/app.js [hunk 10]
```diff
@@ -68,32 +68,34 @@ Hooks.ShowPassword = {
...
   }
 }
 
 // Show progress bar on live navigation and form submits
-topbar.config({barColors: {0: "#29d"}, shadowColor: "rgba(0, 0, 0, .3)"})
-window.addEventListener("phx:page-loading-start", _info => topbar.show(300))
-window.addEventListener("phx:page-loading-stop", _info => topbar.hide())
+topbar.config({ barColors: { 0: "#29d" }, shadowColor: "rgba(0, 0, 0, .3)" })
+window.addEventListener("phx:page-loading-start", (_info) => topbar.show(300))
+window.addEventListener("phx:page-loading-stop", (_info) => topbar.hide())
```
Comment (kuroda) [inline assets/js/app.js]: Please leave this as it is.

**Review by kuroda — CHANGES_REQUESTED** [review]
I add additional (trivial) comments on `app.js`.

File: assets/js/app.js (hunk 4 above)
Comment (kkana6624) [reply assets/js/app.js]: I don't remember directly fixing the relevant part, so I think it might be due to the impact of mix format.

**Review by kuroda — APPROVED** [review]

File: assets/js/app.js (hunk 4 above)
Comment (kuroda) [reply assets/js/app.js]: The `mix format` is an Elixir code formatter and should not touch JavaScript code.

I suspect that you have a code formatter tool installed for JavaScript or Node.js.

There is nothing wrong with using a code formatter tool, but all project members need to use the same tool with the same settings.

**Pull request merged**